torch/nn/parallel/distributed.py

```diff
@@ -599,3 +606,22 @@ def _passing_sync_batchnorm_handle(self, module_copies):
                     assert self.device_type != 'cpu', "SyncBatchNorm layers only work with GPU modules"
                     layer._specify_ddp_gpu_num(
                         len(self.device_ids) if self.device_ids else 1)
+
+    def _check_comm_hook(self, hook):
+        assert callable(hook), "Communication hook must be callable."
+
+        sig = inspect.signature(hook)
+        assert (sig.parameters['state'].annotation != inspect._empty and
+                sig.parameters['bucket'].annotation != inspect._empty and
+                sig.return_annotation != inspect._empty), (
+                    "Communication hook is missing annotations."
```

Comment:
Does this mandate that the user has to type functions they pass to `register_comm_hook`? I'm not sure if PT should take the responsibility of ensuring that the user annotates types on their code, unless it's required (i.e. for TorchScript)

Reply:
I agree. We shouldn't enforce annotations unless it is necessary. Now, we'll check the annotations only if user provides them.